frontend/apps/filemanager/filemanagerfilesearcher.lua

```diff
@@ -110,24 +108,49 @@ function FileSearcher:onShowFileSearch(search_string)
 end
 
 function FileSearcher:doSearch()
-    local dirs, files = self:getList()
-    -- If we have a FileChooser instance, use it, to be able to make use of its natsort cache
-    local results = (self.ui.file_chooser or FileChooser):genItemTable(dirs, files)
-    if #results > 0 then
-        self:showSearchResults(results)
+    local last_search_hash = self.path .. (self.search_string or "") ..
+        tostring(self.case_sensitive) .. tostring(self.include_subfolders) .. tostring(self.include_metadata)
+    local not_cached = self.last_search_hash ~= last_search_hash
+    if not_cached then
+        local Trapper = require("ui/trapper")
+        local info = InfoMessage:new{ text = _("Searching… (tap to cancel)") }
+        UIManager:show(info)
+        UIManager:forceRePaint()
+        local completed, dirs, files, no_metadata_count = Trapper:dismissableRunInSubprocess(function()
+            return self:getList()
+        end, info)
+        if not completed then return end
+        UIManager:close(info)
+        self.last_search_hash = last_search_hash
+        self.no_metadata_count = no_metadata_count
+        -- Cannot do this in getList() within Trapper (cannot serialize function)
+        local collate = FileChooser:getCollate()
+        for i, v in ipairs(dirs) do
+            local f, fullpath, attributes = unpack(v)
+            dirs[i] = FileChooser:getListItem(nil, f, fullpath, attributes, collate)
+        end
+        for i, v in ipairs(files) do
+            local f, fullpath, attributes = unpack(v)
+            files[i] = FileChooser:getListItem(nil, f, fullpath, attributes, collate)
+        end
+        -- If we have a FileChooser instance, use it, to be able to make use of its natsort cache
+        self.search_results = (self.ui.file_chooser or FileChooser):genItemTable(dirs, files)
+    end
+    if #self.search_results > 0 then
+        self:onShowSearchResults(not_cached)
     else
         self:showSearchResultsMessage(true)
     end
 end
 
 function FileSearcher:getList()
-    self.no_metadata_count = 0
+    self.no_metadata_count = 0 -- in subprocess
```

Comment:
`-- in subprocess` reads vague, I don't know what you mean when reading it.

Reply:
The comment explains the following line:
https://github.com/koreader/koreader/blob/b9e6519c97fec4a65b3a19b6a7f179ab5a84a420/frontend/apps/filemanager/filemanagerfilesearcher.lua#L204
It's another `self.no_metadata_count`.

Reply:
So, a better comment maybe?:
`-- will be updated in doSearch() with result from subprocess`